src/awkward/behaviors/string.py

```diff
@@ -249,7 +249,7 @@ def register(behavior):
     behavior["__typestr__", "char"] = "char"
 
     behavior["bytestring"] = ByteStringBehavior
-    behavior["__typestr__", "bytestring"] = "bytes"
+    behavior["__typestr__", "bytestring"] = "byte"
```

Comment:
This is the convention that was fixed for a while, it was just wrong in a few places in the codebase. We might even find that some of the "lost strings" issues are fixed now.

The reason for singular is the same as the reason why "int" or "float" (singular) is used for an array of integers or floats. Or "string" for an array of strings. This is a NumpyArray in which each element is a byte, not a ListArray in which every element is a bytestring.

Similarly for "char" in "string" arrays.

Reply:
The second to last (many byte strings being "byte") is wrong and you found a mistake that I added.

```python
<Array [b'hi', b'hello'] type='2 * byte'>
```

is indeed wrong. I know what mistake I made: our parameter names are `"byte"` and `"bytestring"`, but the typestr for bytestring needs to be `"bytes"` to satisfy the Datashape language.

I think you found a case that should have remained `"bytes"`, because it's not a parameter. It's a typestr.